tests/acceptance/17_packages/10_new/unsafe/the_great_package_test_generator.py

```diff
@@ -0,0 +1,540 @@
+                      "PF64_2A64V2",
+                      "PF32_1A32V1",
+                      "PF32_2A32V2",
+                      "PR         ",
+                      "PR    A64  ",
+                      "PR    A32  ",
+                      "PR       V1",
+                      "PR       V2",
+                      "PR       VL",
+                      "PR    A64V1",
+                      "PR    A32V1",
+                      "PR    A64V2",
+                      "PR    A32V2",
+                      "PR    A64VL",
+                      "PR    A32VL",
+                      "D          ",
+                      "D     A64  ",
+                      "D     A32  ",
+                      "D        V1",
+                      "D        V2",
+                      "D     A64V1",
+                      "D     A32V1",
+                      "D     A64V2",
+                      "D     A32V2",
+                    ]
+
+    promises = []
+    for text in text_promises:
+        promise = {}
+        if text[0] == "P":
+            promise["policy"] = "present"
+            if text[1] == "F":
+                promise["type"] = "file"
+                if text[2:4] == "64":
+                    promise["file_arch"] = "64_bit"
+                else:
+                    promise["file_arch"] = "32_bit"
+                if text[5] == "1":
+                    promise["file_version"] = "1"
+                else:
+                    promise["file_version"] = "2"
+            else:
+                promise["type"] = "repo"
+        else:
+            promise["policy"] = "absent"
+        if text[6:9] == "A64":
+            promise["arch"] = "64_bit"
+        if text[6:9] == "A32":
+            promise["arch"] = "32_bit"
+        if text[9:11] == "VL":
+            promise["version"] = "latest"
+        elif text[9] == "V":
+            promise["version"] = text[10]
+
+        promises.append(promise)
+
+    return promises
+
+
+# Given the transitions between the two version, determine if the change of one
+# architecture package will trigger a change in the other. Only one of the two
+# inputs can be different, but the returned value may cause both to be
+# different. This is only relevant for "repo" style promises, since "file"
+# promises will never touch more than one package at a time.
+def resolve_arch_conflicts(from_64, from_32, to_64, to_32):
+    if from_64 != to_64:
+        if from_32 != "0":
+            if to_64 >= from_32:
+                to_32 = to_64
+            else:
+                to_32 = "0"
+    elif from_32 != to_32:
+        if from_64 != "0":
+            if to_32 >= from_64:
+                to_64 = to_32
+            else:
+                to_64 = "0"
+
+    return (to_64, to_32)
+
+
+# Simulate, with the given promise, what the state of the system would be,
+# if we started from the versions in from_64 and from_32.
+def simulate_promise(promise, from_64, from_32):
+    to_64 = from_64
+    to_32 = from_32
+
```

Comment:
I think this would all be *hugely* easier to read if you started with
```
arch, version = promise.get('arch'), promise.get('version')
```
and similar for anything else you `.get()` repeatedly otherwise.  It would incidentally also run faster.